README.md

```diff
@@ -86,20 +84,124 @@ code because of all the basic column-renamings and the full listing of health co
+parse a proto-schema provided by DAP into usable code for ingest in the form of a Terra JSON Schema output.
+
+
+#### Creating Outfiles 
+DAP's long term future would include an official TDR<->Workspace integration but we are currently still building that out.
+To give ourselves some breathing room, we've written a [script](./hack/convert-output-to-tsv.py) to convert
+the outputs of our transformation pipeline into workspace-compatible TSVs. 
 The script takes two positional arguments:
 1. Path to the top-level directory containing the outputs of the transformation pipeline
 2. Path to a directory where converted TSVs should be written
+There are additional arguments to that script to add some flexibility to the tool:
+* Can take in a list of tables to process (currently looks for all tables)
+* Can output primary keys with a modified column name for Firecloud compatibility: *entity:{table_name}_id*
+* Call Format:
+    * `python convert-output-to-tsv.py {transform GS path} {tsv outfile local path} -d`
+    * `python convert-output-to-tsv.py gs://broad-dsp-monster/weekly_refresh/20210525/transform /Users/DATA/DAP/tsv -d`
+
+#### Validation
+We have a set of validation tests that we run to QA new TSVs. These validation tests are used to look for specific data
+issues that were discovered in creating the curated 2020 refresh in coordination with the DAP team.
+
+## Refreshing Data
+### Running the pipelines on GCS
+While it is possible to manually process the DAP data on our local machines, it had become too cumbersome to do so once we 
+crossed the threshold of around 10k+ participants. We pushed our extraction and transformation logic out to the cloud and 
+are now running our refreshes in *dev_mode* using the dataflow_beam runner.
+The Dagster project we added for DAP includes 3 different modes of operation:
+1. `local_mode` uses local_beam_runner
+1. `dev_mode` uses dataflow_beam_runner
+1. `prod_mode` uses dataflow_beam_runner
+
+For now, we can call individual pipelines manually with any runner and then call the tsv script to write files locally.
+The general structure for the SBT call to kick off one of our pipelines:
+* `sbt "{scala_project} {target_class} {sbt parameters} {dataflow parameters}"`
+* Here is sample an sbt for the Environment extraction pipeline:
+    >sbt "dog-aging-hles-extraction/runMain org.broadinstitute.monster.dap.environment.EnvironmentExtractionPipeline 
+    --apiToken=foo --pullDataDictionaries=false --outputPrefix=bar --runner=dataflow --project=baz --region=us-central1
+    --workerMachineType=n1-standard-1 --autoscalingAlgorithm=THROUGHPUT_BASED --numWorkers=4 --maxNumWorkers=8 
+    --experiments=shuffle_mode=service --endTime=qux"
+
+Alternatively we can also execute a Dagster pipeline which will run all the pipelines and write local TSVs and will 
+use a runner based on the mode.
+
+### Manual Refresh (Dagster)
+1. Navigate to the *orhcestration* subdirectory
+2. Poetry installation
```

Comment:
⛏️ This isn't really installing poetry, but rather setting up the DAP venv. Perhaps:

```suggestion
2. Setup python virtual environment
```

Reply:
I think it'd also be worth adding a note somewhere in this section to indicate that the dev should install poetry before getting started